src/functions-and-arrays.js

```diff
@@ -1,41 +1,143 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+    } else if (num2 > num1) {
+      return num2;
+    }
+      return num1;
+    }
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(words) {
+    if (!words.length) {
+    return null;
+  }
+  let longest = ' ';
+  for (let i = 0; i < words.length; i++) {
+    if (words[i].length > longest.length) {
+    longest = words[i]; 
+    }
+  }
+  return longest;
+}
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbers) {
+  if (!numbers.length) {
+    return 0;
+    } 
+    let total = 0;
+    for (let i = 0; i < numbers.length; i++) {
+    total += numbers[i];
+    }
+  return total;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
+
+function sum(mixedArr) {
+  let mixedSumOne = 0;
+  let mixedSumTwo = 0;
+  let mixedSumTotal = 0;
+
+  if (!mixedArr.length) {
+    return 0;
+    
+} for (let i = 0; i < mixedArr.length; i++) {
+
+  if (typeof mixedArr[i] === 'object' || typeof mixedArr[i] === 'array') {
+    throw "Unsupported data type sir or ma'am";
+  }
+  if (typeof mixedArr[i] === 'number') {
+    mixedSumOne = mixedArr[i];
+    mixedSumTotal += mixedSumOne;
+  } else if (typeof mixedArr[i] === 'string') {
+    mixedSumTwo = mixedArr[i].length;
+    mixedSumTotal += mixedSumTwo;
+  } else if (typeof mixedArr[i] === 'boolean') {
+    if (mixedArr[i] === true) {
+      mixedSumTotal += 1
+    } else {
+      mixedSumTotal += 0
+  } 
+}
+}
+  return mixedSumTotal;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+  if (!numbersAvg.length) {
+    return null;
+  }
+  let average = 0;
+  let total = 0;
+  for (let i = 0; i < numbersAvg.length; i++) {
+    total += numbersAvg[i]
+    average = total / numbersAvg.length
```

Comment:
esse `average` fica melhor se ficar fora do `for`. Assim ele faria a conta apenas uma vez